### PR TITLE
【PIR API adaptor No.1、20、103、104、120】 Migrate L1Loss/BCELoss/HSigmoidLoss/SmoothL1Loss/KLDivLoss into pir

### PR DESCRIPTION
**Pull request by DrRyanHuang**
### PR types

Others

### PR changes

APIs

### Description
- https://github.com/PaddlePaddle/Paddle/issues/58067
- 

- paddle.nn.L1Loss
- paddle.nn.BCELoss
- paddle.nn.HSigmoidLoss
- paddle.nn.SmoothL1Loss
- paddle.nn.KLDivLoss

------
- `test/legacy_test/test_l1_loss.py` 文件下
`self.assertTrue('aaa' in result3.name)` 不支持pir model，于是将 `TestClassL1Loss.run_static` 和 `TestFunctionalL1Loss.run_static` 关闭

- `test/legacy_test/test_hsigmoid_op.py` 没有开启 `test_hs_grad_with_sparse`，原因是 调用的 `paddle.static.nn.embedding` 和`base.DataFeeder` 应该没有适配pir

### CODE REVIEW

File: test/legacy_test/test_l1_loss.py [hunk 1]
```diff
@@ -133,45 +144,51 @@ def run_imperative(self):
...
-        self.assertTrue('aaa' in result3.name)
+            expected = np.mean(np.abs(self.input_np - self.label_np))
+            np.testing.assert_allclose(static_result[0], expected, rtol=1e-05)
+            expected = np.sum(np.abs(self.input_np - self.label_np))
+            np.testing.assert_allclose(static_result[1], expected, rtol=1e-05)
+            expected = np.abs(self.input_np - self.label_np)
+            np.testing.assert_allclose(static_result[2], expected, rtol=1e-05)
+
+            if in_pir_mode():
+                self.assertTrue('aaa' in result3)
```
Comment (0x45f) [inline test/legacy_test/test_l1_loss.py]: pir下result3是OpResult，不能'aaa' in result3 这样检查。pir下不要检查了，只在动态图下做name 的检查吧

File: test/legacy_test/test_hsigmoid_op.py [hunk 2]
```diff
@@ -358,6 +360,7 @@ def training_test(self, is_sparse):
                 result.append(loss_val)
         return result
 
+    @test_with_pir_api
```
Comment (0x45f) [inline test/legacy_test/test_hsigmoid_op.py]: 这个单测先跳过吧，里面调用的paddle.static.nn.embedding和base.DataFeeder应该没有适配pir。跳过了单测记得在pr描述里标记一下跳过的原因

File: test/legacy_test/test_hsigmoid_op.py [hunk 3]
```diff
@@ -612,17 +618,16 @@ def test_static_api(self):
             if self.is_custom:
                 feed_dict["path_code"] = self.path_code_np
                 feed_dict["path_table"] = self.path_table_np
-            ret1, ret2 = exe.run(
-                train_program, feed=feed_dict, fetch_list=[out1, out2]
-            )
+            ret1, ret2 = exe.run(feed=feed_dict, fetch_list=[out1, out2])
```
Comment (0x45f) [inline test/legacy_test/test_hsigmoid_op.py]: 为什么把上面的exe.run(startup_program)删掉了？这个不能删吧

File: test/legacy_test/test_bce_loss.py [hunk 4]
```diff
@@ -55,6 +56,7 @@ def test_static_layer(
     return static_result
 
 
+@test_with_pir_api
```
Comment (0x45f) [inline test/legacy_test/test_bce_loss.py]: 这里不需要加装饰器~

**Comment by DrRyanHuang** [comment]
目前需要等待 https://github.com/PaddlePaddle/Paddle/pull/58876 先合入

**Comment by MarioLulab** [comment]
nn.initializer.Bilinear/Assign 已迁移至 pir 下：https://github.com/PaddlePaddle/Paddle/pull/60114
此 pr 可以继续推进了~

**Comment by MarioLulab** [comment]
HsigmoidLossOp::Vjp遇到报错，报错栈为
```bash
2023-12-20 03:24:01 I1219 19:24:00.857872 24887 pass.cc:38] --- detected [0] subgraphs!
2023-12-20 03:24:01 test_hsigmoid_op failed
2023-12-20 03:24:01  .............E..
2023-12-20 03:24:01 ======================================================================
2023-12-20 03:24:01 ERROR: test_check_grad (test_hsigmoid_op.TestHSigmoidOpWithCostumTreeWithoutBias)
2023-12-20 03:24:01 ----------------------------------------------------------------------
2023-12-20 03:24:01 Traceback (most recent call last):
2023-12-20 03:24:01   File "/workspace/Paddle/build/test/legacy_test/test_hsigmoid_op.py", line 490, in test_check_grad
2023-12-20 03:24:01     self.check_grad(
2023-12-20 03:24:01   File "/workspace/Paddle/build/python/op_test.py", line 2969, in check_grad
2023-12-20 03:24:01     self.check_grad_with_place(
2023-12-20 03:24:01   File "/workspace/Paddle/build/python/op_test.py", line 3288, in check_grad_with_place
2023-12-20 03:24:01     pir_grad = self._get_ir_gradient(
2023-12-20 03:24:01   File "/workspace/Paddle/build/python/op_test.py", line 3814, in _get_ir_gradient
2023-12-20 03:24:01     grad_inputs = ir_grad(
2023-12-20 03:24:01   File "/workspace/Paddle/build/python/paddle/autograd/ir_backward.py", line 1077, in grad
2023-12-20 03:24:01     input_grad = calc_gradient(outputs, inputs, grad_outputs, no_grad_set)
2023-12-20 03:24:01   File "/workspace/Paddle/build/python/paddle/autograd/ir_backward.py", line 962, in calc_gradient
2023-12-20 03:24:01     input_to_inputgrad_map = calc_gradient_helper(
2023-12-20 03:24:01   File "/workspace/Paddle/build/python/paddle/autograd/ir_backward.py", line 901, in calc_gradient_helper
2023-12-20 03:24:01     append_backward_ops(
2023-12-20 03:24:01   File "/workspace/Paddle/build/python/paddle/autograd/ir_backward.py", line 759, in append_backward_ops
2023-12-20 03:24:01     input_grads = paddle.framework.core.call_vjp(
2023-12-20 03:24:01 ValueError: 
2023-12-20 03:24:01 
2023-12-20 03:24:01 0   paddle::dialect::VjpInterface::Vjp(pir::Operation*, std::vector<std::vector<pir::Value, std::allocator<pir::Value> >, std::allocator<std::vector<pir::Value, std::allocator<pir::Value> > > > const&, std::vector<std::vector<pir::OpResult, std::allocator<pir::OpResult> >, std::allocator<std::vector<pir::OpResult, std::allocator<pir::OpResult> > > > const&, std::vector<std::vector<pir::OpResult, std::allocator<pir::OpResult> >, std::allocator<std::vector<pir::OpResult, std::allocator<pir::OpResult> > > > const&, std::vector<std::vector<bool, std::allocator<bool> >, std::allocator<std::vector<bool, std::allocator<bool> > > > const&)
2023-12-20 03:24:01 1   paddle::dialect::VjpInterface::Model<paddle::dialect::HsigmoidLossOp>::Vjp(pir::Operation*, std::vector<std::vector<pir::Value, std::allocator<pir::Value> >, std::allocator<std::vector<pir::Value, std::allocator<pir::Value> > > > const&, std::vector<std::vector<pir::OpResult, std::allocator<pir::OpResult> >, std::allocator<std::vector<pir::OpResult, std::allocator<pir::OpResult> > > > const&, std::vector<std::vector<pir::Value, std::allocator<pir::Value> >, std::allocator<std::vector<pir::Value, std::allocator<pir::Value> > > > const&, std::vector<std::vector<bool, std::allocator<bool> >, std::allocator<std::vector<bool, std::allocator<bool> > > > const&)
2023-12-20 03:24:01 2   paddle::dialect::HsigmoidLossOp::Vjp(pir::Operation*, std::vector<std::vector<pir::Value, std::allocator<pir::Value> >, std::allocator<std::vector<pir::Value, std::allocator<pir::Value> > > > const&, std::vector<std::vector<pir::OpResult, std::allocator<pir::OpResult> >, std::allocator<std::vector<pir::OpResult, std::allocator<pir::OpResult> > > > const&, std::vector<std::vector<pir::Value, std::allocator<pir::Value> >, std::allocator<std::vector<pir::Value, std::allocator<pir::Value> > > > const&, std::vector<std::vector<bool, std::allocator<bool> >, std::allocator<std::vector<bool, std::allocator<bool> > > > const&)
2023-12-20 03:24:01 3   paddle::primitive::hsigmoid_loss_vjp(paddle::Tensor const&, paddle::Tensor const&, paddle::Tensor const&, paddle::optional<paddle::Tensor> const&, paddle::optional<paddle::Tensor> const&, paddle::optional<paddle::Tensor> const&, paddle::Tensor const&, paddle::Tensor const&, int, bool, std::vector<std::vector<bool, std::allocator<bool> >, std::allocator<std::vector<bool, std::allocator<bool> > > > const&)
2023-12-20 03:24:01 4   std::tuple<paddle::Tensor, paddle::Tensor, paddle::Tensor> paddle::primitive::backend::hsigmoid_loss_grad<paddle::primitive::LazyTensor>(paddle::Tensor const&, paddle::Tensor const&, paddle::Tensor const&, paddle::optional<paddle::Tensor> const&, paddle::optional<paddle::Tensor> const&, paddle::optional<paddle::Tensor> const&, paddle::Tensor const&, paddle::Tensor const&, int, bool)
2023-12-20 03:24:01 5   paddle::dialect::hsigmoid_loss_grad(pir::Value const&, pir::Value const&, pir::Value const&, paddle::optional<pir::Value> const&, paddle::optional<pir::Value> const&, paddle::optional<pir::Value> const&, pir::Value const&, pir::Value const&, int, bool)
2023-12-20 03:24:01 6   paddle::dialect::HsigmoidLossGradOp::Build(pir::Builder&, pir::OperationArgument&, pir::Value, pir::Value, pir::Value, pir::Value, pir::Value, pir::Value, pir::Value, pir::Value, int, bool)
2023-12-20 03:24:01 7   paddle::dialect::IrMetaTensor::share_meta(phi::MetaTensor const&)
2023-12-20 03:24:01 8   phi::enforce::EnforceNotMet::EnforceNotMet(common::ErrorSummary const&, char const*, int)
2023-12-20 03:24:01 9   phi::enforce::GetCurrentTraceBackString[abi:cxx11](bool)
2023-12-20 03:24:01 
2023-12-20 03:24:01 ----------------------
2023-12-20 03:24:01 Error Message Summary:
2023-12-20 03:24:01 ----------------------
2023-12-20 03:24:01 InvalidArgumentError: The current MetaTensor is not initialized.
2023-12-20 03:24:01   [Hint: Expected meta_tensor.initialized() == true, but received meta_tensor.initialized():0 != true:1.] (at ../paddle/fluid/pir/dialect/operator/ir/ir_meta_tensor.cc:22)
```

**Comment by 0x45f** [comment]
> HsigmoidLossOp::Vjp遇到报错，报错栈为
> 
> ```shell
> 2023-12-20 03:24:01 I1219 19:24:00.857872 24887 pass.cc:38] --- detected [0] subgraphs!
> 2023-12-20 03:24:01 test_hsigmoid_op failed
> 2023-12-20 03:24:01  .............E..
> 2023-12-20 03:24:01 ======================================================================
> 2023-12-20 03:24:01 ERROR: test_check_grad (test_hsigmoid_op.TestHSigmoidOpWithCostumTreeWithoutBias)
> 2023-12-20 03:24:01 ----------------------------------------------------------------------
> 2023-12-20 03:24:01 Traceback (most recent call last):
> 2023-12-20 03:24:01   File "/workspace/Paddle/build/test/legacy_test/test_hsigmoid_op.py", line 490, in test_check_grad
> 2023-12-20 03:24:01     self.check_grad(
> 2023-12-20 03:24:01   File "/workspace/Paddle/build/python/op_test.py", line 2969, in check_grad
> 2023-12-20 03:24:01     self.check_grad_with_place(
> 2023-12-20 03:24:01   File "/workspace/Paddle/build/python/op_test.py", line 3288, in check_grad_with_place
> 2023-12-20 03:24:01     pir_grad = self._get_ir_gradient(
> 2023-12-20 03:24:01   File "/workspace/Paddle/build/python/op_test.py", line 3814, in _get_ir_gradient
> 2023-12-20 03:24:01     grad_inputs = ir_grad(
> 2023-12-20 03:24:01   File "/workspace/Paddle/build/python/paddle/autograd/ir_backward.py", line 1077, in grad
> 2023-12-20 03:24:01     input_grad = calc_gradient(outputs, inputs, grad_outputs, no_grad_set)
> 2023-12-20 03:24:01   File "/workspace/Paddle/build/python/paddle/autograd/ir_backward.py", line 962, in calc_gradient
> 2023-12-20 03:24:01     input_to_inputgrad_map = calc_gradient_helper(
> 2023-12-20 03:24:01   File "/workspace/Paddle/build/python/paddle/autograd/ir_backward.py", line 901, in calc_gradient_helper
> 2023-12-20 03:24:01     append_backward_ops(
> 2023-12-20 03:24:01   File "/workspace/Paddle/build/python/paddle/autograd/ir_backward.py", line 759, in append_backward_ops
> 2023-12-20 03:24:01     input_grads = paddle.framework.core.call_vjp(
> 2023-12-20 03:24:01 ValueError: 
> 2023-12-20 03:24:01 
> 2023-12-20 03:24:01 0   paddle::dialect::VjpInterface::Vjp(pir::Operation*, std::vector<std::vector<pir::Value, std::allocator<pir::Value> >, std::allocator<std::vector<pir::Value, std::allocator<pir::Value> > > > const&, std::vector<std::vector<pir::OpResult, std::allocator<pir::OpResult> >, std::allocator<std::vector<pir::OpResult, std::allocator<pir::OpResult> > > > const&, std::vector<std::vector<pir::OpResult, std::allocator<pir::OpResult> >, std::allocator<std::vector<pir::OpResult, std::allocator<pir::OpResult> > > > const&, std::vector<std::vector<bool, std::allocator<bool> >, std::allocator<std::vector<bool, std::allocator<bool> > > > const&)
> 2023-12-20 03:24:01 1   paddle::dialect::VjpInterface::Model<paddle::dialect::HsigmoidLossOp>::Vjp(pir::Operation*, std::vector<std::vector<pir::Value, std::allocator<pir::Value> >, std::allocator<std::vector<pir::Value, std::allocator<pir::Value> > > > const&, std::vector<std::vector<pir::OpResult, std::allocator<pir::OpResult> >, std::allocator<std::vector<pir::OpResult, std::allocator<pir::OpResult> > > > const&, std::vector<std::vector<pir::Value, std::allocator<pir::Value> >, std::allocator<std::vector<pir::Value, std::allocator<pir::Value> > > > const&, std::vector<std::vector<bool, std::allocator<bool> >, std::allocator<std::vector<bool, std::allocator<bool> > > > const&)
> 2023-12-20 03:24:01 2   paddle::dialect::HsigmoidLossOp::Vjp(pir::Operation*, std::vector<std::vector<pir::Value, std::allocator<pir::Value> >, std::allocator<std::vector<pir::Value, std::allocator<pir::Value> > > > const&, std::vector<std::vector<pir::OpResult, std::allocator<pir::OpResult> >, std::allocator<std::vector<pir::OpResult, std::allocator<pir::OpResult> > > > const&, std::vector<std::vector<pir::Value, std::allocator<pir::Value> >, std::allocator<std::vector<pir::Value, std::allocator<pir::Value> > > > const&, std::vector<std::vector<bool, std::allocator<bool> >, std::allocator<std::vector<bool, std::allocator<bool> > > > const&)
> 2023-12-20 03:24:01 3   paddle::primitive::hsigmoid_loss_vjp(paddle::Tensor const&, paddle::Tensor const&, paddle::Tensor const&, paddle::optional<paddle::Tensor> const&, paddle::optional<paddle::Tensor> const&, paddle::optional<paddle::Tensor> const&, paddle::Tensor const&, paddle::Tensor const&, int, bool, std::vector<std::vector<bool, std::allocator<bool> >, std::allocator<std::vector<bool, std::allocator<bool> > > > const&)
> 2023-12-20 03:24:01 4   std::tuple<paddle::Tensor, paddle::Tensor, paddle::Tensor> paddle::primitive::backend::hsigmoid_loss_grad<paddle::primitive::LazyTensor>(paddle::Tensor const&, paddle::Tensor const&, paddle::Tensor const&, paddle::optional<paddle::Tensor> const&, paddle::optional<paddle::Tensor> const&, paddle::optional<paddle::Tensor> const&, paddle::Tensor const&, paddle::Tensor const&, int, bool)
> 2023-12-20 03:24:01 5   paddle::dialect::hsigmoid_loss_grad(pir::Value const&, pir::Value const&, pir::Value const&, paddle::optional<pir::Value> const&, paddle::optional<pir::Value> const&, paddle::optional<pir::Value> const&, pir::Value const&, pir::Value const&, int, bool)
> 2023-12-20 03:24:01 6   paddle::dialect::HsigmoidLossGradOp::Build(pir::Builder&, pir::OperationArgument&, pir::Value, pir::Value, pir::Value, pir::Value, pir::Value, pir::Value, pir::Value, pir::Value, int, bool)
> 2023-12-20 03:24:01 7   paddle::dialect::IrMetaTensor::share_meta(phi::MetaTensor const&)
> 2023-12-20 03:24:01 8   phi::enforce::EnforceNotMet::EnforceNotMet(common::ErrorSummary const&, char const*, int)
> 2023-12-20 03:24:01 9   phi::enforce::GetCurrentTraceBackString[abi:cxx11](bool)
> 2023-12-20 03:24:01 
> 2023-12-20 03:24:01 ----------------------
> 2023-12-20 03:24:01 Error Message Summary:
> 2023-12-20 03:24:01 ----------------------
> 2023-12-20 03:24:01 InvalidArgumentError: The current MetaTensor is not initialized.
> 2023-12-20 03:24:01   [Hint: Expected meta_tensor.initialized() == true, but received meta_tensor.initialized():0 != true:1.] (at ../paddle/fluid/pir/dialect/operator/ir/ir_meta_tensor.cc:22)
> ```

内部相关同学在解决这个问题了~

**Comment by changeyoung98** [comment]
> HsigmoidLossOp::Vjp遇到报错，报错栈为
> 
> ```shell
> 2023-12-20 03:24:01 I1219 19:24:00.857872 24887 pass.cc:38] --- detected [0] subgraphs!
> 2023-12-20 03:24:01 test_hsigmoid_op failed
> 2023-12-20 03:24:01  .............E..
> 2023-12-20 03:24:01 ======================================================================
> 2023-12-20 03:24:01 ERROR: test_check_grad (test_hsigmoid_op.TestHSigmoidOpWithCostumTreeWithoutBias)
> 2023-12-20 03:24:01 ----------------------------------------------------------------------
> 2023-12-20 03:24:01 Traceback (most recent call last):
> 2023-12-20 03:24:01   File "/workspace/Paddle/build/test/legacy_test/test_hsigmoid_op.py", line 490, in test_check_grad
> 2023-12-20 03:24:01     self.check_grad(
> 2023-12-20 03:24:01   File "/workspace/Paddle/build/python/op_test.py", line 2969, in check_grad
> 2023-12-20 03:24:01     self.check_grad_with_place(
> 2023-12-20 03:24:01   File "/workspace/Paddle/build/python/op_test.py", line 3288, in check_grad_with_place
> 2023-12-20 03:24:01     pir_grad = self._get_ir_gradient(
> 2023-12-20 03:24:01   File "/workspace/Paddle/build/python/op_test.py", line 3814, in _get_ir_gradient
> 2023-12-20 03:24:01     grad_inputs = ir_grad(
> 2023-12-20 03:24:01   File "/workspace/Paddle/build/python/paddle/autograd/ir_backward.py", line 1077, in grad
> 2023-12-20 03:24:01     input_grad = calc_gradient(outputs, inputs, grad_outputs, no_grad_set)
> 2023-12-20 03:24:01   File "/workspace/Paddle/build/python/paddle/autograd/ir_backward.py", line 962, in calc_gradient
> 2023-12-20 03:24:01     input_to_inputgrad_map = calc_gradient_helper(
> 2023-12-20 03:24:01   File "/workspace/Paddle/build/python/paddle/autograd/ir_backward.py", line 901, in calc_gradient_helper
> 2023-12-20 03:24:01     append_backward_ops(
> 2023-12-20 03:24:01   File "/workspace/Paddle/build/python/paddle/autograd/ir_backward.py", line 759, in append_backward_ops
> 2023-12-20 03:24:01     input_grads = paddle.framework.core.call_vjp(
> 2023-12-20 03:24:01 ValueError: 
> 2023-12-20 03:24:01 
> 2023-12-20 03:24:01 0   paddle::dialect::VjpInterface::Vjp(pir::Operation*, std::vector<std::vector<pir::Value, std::allocator<pir::Value> >, std::allocator<std::vector<pir::Value, std::allocator<pir::Value> > > > const&, std::vector<std::vector<pir::OpResult, std::allocator<pir::OpResult> >, std::allocator<std::vector<pir::OpResult, std::allocator<pir::OpResult> > > > const&, std::vector<std::vector<pir::OpResult, std::allocator<pir::OpResult> >, std::allocator<std::vector<pir::OpResult, std::allocator<pir::OpResult> > > > const&, std::vector<std::vector<bool, std::allocator<bool> >, std::allocator<std::vector<bool, std::allocator<bool> > > > const&)
> 2023-12-20 03:24:01 1   paddle::dialect::VjpInterface::Model<paddle::dialect::HsigmoidLossOp>::Vjp(pir::Operation*, std::vector<std::vector<pir::Value, std::allocator<pir::Value> >, std::allocator<std::vector<pir::Value, std::allocator<pir::Value> > > > const&, std::vector<std::vector<pir::OpResult, std::allocator<pir::OpResult> >, std::allocator<std::vector<pir::OpResult, std::allocator<pir::OpResult> > > > const&, std::vector<std::vector<pir::Value, std::allocator<pir::Value> >, std::allocator<std::vector<pir::Value, std::allocator<pir::Value> > > > const&, std::vector<std::vector<bool, std::allocator<bool> >, std::allocator<std::vector<bool, std::allocator<bool> > > > const&)
> 2023-12-20 03:24:01 2   paddle::dialect::HsigmoidLossOp::Vjp(pir::Operation*, std::vector<std::vector<pir::Value, std::allocator<pir::Value> >, std::allocator<std::vector<pir::Value, std::allocator<pir::Value> > > > const&, std::vector<std::vector<pir::OpResult, std::allocator<pir::OpResult> >, std::allocator<std::vector<pir::OpResult, std::allocator<pir::OpResult> > > > const&, std::vector<std::vector<pir::Value, std::allocator<pir::Value> >, std::allocator<std::vector<pir::Value, std::allocator<pir::Value> > > > const&, std::vector<std::vector<bool, std::allocator<bool> >, std::allocator<std::vector<bool, std::allocator<bool> > > > const&)
> 2023-12-20 03:24:01 3   paddle::primitive::hsigmoid_loss_vjp(paddle::Tensor const&, paddle::Tensor const&, paddle::Tensor const&, paddle::optional<paddle::Tensor> const&, paddle::optional<paddle::Tensor> const&, paddle::optional<paddle::Tensor> const&, paddle::Tensor const&, paddle::Tensor const&, int, bool, std::vector<std::vector<bool, std::allocator<bool> >, std::allocator<std::vector<bool, std::allocator<bool> > > > const&)
> 2023-12-20 03:24:01 4   std::tuple<paddle::Tensor, paddle::Tensor, paddle::Tensor> paddle::primitive::backend::hsigmoid_loss_grad<paddle::primitive::LazyTensor>(paddle::Tensor const&, paddle::Tensor const&, paddle::Tensor const&, paddle::optional<paddle::Tensor> const&, paddle::optional<paddle::Tensor> const&, paddle::optional<paddle::Tensor> const&, paddle::Tensor const&, paddle::Tensor const&, int, bool)
> 2023-12-20 03:24:01 5   paddle::dialect::hsigmoid_loss_grad(pir::Value const&, pir::Value const&, pir::Value const&, paddle::optional<pir::Value> const&, paddle::optional<pir::Value> const&, paddle::optional<pir::Value> const&, pir::Value const&, pir::Value const&, int, bool)
> 2023-12-20 03:24:01 6   paddle::dialect::HsigmoidLossGradOp::Build(pir::Builder&, pir::OperationArgument&, pir::Value, pir::Value, pir::Value, pir::Value, pir::Value, pir::Value, pir::Value, pir::Value, int, bool)
> 2023-12-20 03:24:01 7   paddle::dialect::IrMetaTensor::share_meta(phi::MetaTensor const&)
> 2023-12-20 03:24:01 8   phi::enforce::EnforceNotMet::EnforceNotMet(common::ErrorSummary const&, char const*, int)
> 2023-12-20 03:24:01 9   phi::enforce::GetCurrentTraceBackString[abi:cxx11](bool)
> 2023-12-20 03:24:01 
> 2023-12-20 03:24:01 ----------------------
> 2023-12-20 03:24:01 Error Message Summary:
> 2023-12-20 03:24:01 ----------------------
> 2023-12-20 03:24:01 InvalidArgumentError: The current MetaTensor is not initialized.
> 2023-12-20 03:24:01   [Hint: Expected meta_tensor.initialized() == true, but received meta_tensor.initialized():0 != true:1.] (at ../paddle/fluid/pir/dialect/operator/ir/ir_meta_tensor.cc:22)
> ```

此问题已解决：https://github.com/PaddlePaddle/Paddle/pull/60264 ，可拉取最新代码进行测试。

File: test/legacy_test/test_bce_loss.py [hunk 5]
```diff
@@ -286,16 +289,17 @@ def init_test_cast(self):
...
-        self.check_grad(['X'], 'Out')
+        self.check_grad(['X'], 'Out', check_pir=True)
 
     def init_test_dtype(self):
         self.dtype = np.float16
 
 
 class TestBceLossOpStaticFP16(unittest.TestCase):
+    @test_with_pir_api
     def test_fp16(self):
```
Comment (MarioLulab) [inline test/legacy_test/test_bce_loss.py:303]: 在正式组网前加上一句判断
```python
if not core.is_compiled_with_cuda():
    return
```

否则会导致 cpu 环境下组网时找不到 fp16 数据类型

**Review by MarioLulab — COMMENTED** [review]
还需要 pre-commit 一下

File: test/legacy_test/test_l1_loss.py [hunk 6]
```diff
@@ -133,45 +140,48 @@ def run_imperative(self):
...
-        np.testing.assert_allclose(static_result[2], expected, rtol=1e-05)
-        self.assertTrue('aaa' in result3.name)
+            expected = np.mean(np.abs(self.input_np - self.label_np))
+            np.testing.assert_allclose(static_result[0], expected, rtol=1e-05)
+            expected = np.sum(np.abs(self.input_np - self.label_np))
+            np.testing.assert_allclose(static_result[1], expected, rtol=1e-05)
+            expected = np.abs(self.input_np - self.label_np)
+            np.testing.assert_allclose(static_result[2], expected, rtol=1e-05)
+
+            self.assertTrue('aaa' in result3.name)
```
Comment (MarioLulab) [inline test/legacy_test/test_l1_loss.py]: pir 模式下当前场景还不支持通过 ".name" 的方式获取非 persistable 的 Value 名称。所以这里可以加一个是否处于 pir 模式的分支判断，如果不处于 pir 模式下才去获取 result3.name

File: test/legacy_test/test_l1_loss.py [hunk 7]
```diff
@@ -43,42 +44,48 @@ def run_imperative(self):
...
+
+            expected = np.mean(np.abs(self.input_np - self.label_np))
+            np.testing.assert_allclose(static_result[0], expected, rtol=1e-05)
+            expected = np.sum(np.abs(self.input_np - self.label_np))
+            np.testing.assert_allclose(static_result[1], expected, rtol=1e-05)
+            expected = np.abs(self.input_np - self.label_np)
+            np.testing.assert_allclose(static_result[2], expected, rtol=1e-05)
 
-        self.assertTrue('aaa' in y.name)
+            self.assertTrue('aaa' in y.name)
```
Comment (MarioLulab) [inline test/legacy_test/test_l1_loss.py]: pir 模式下当前场景还不支持通过 ".name" 的方式获取非 persistable 的 Value 名称。所以这里可以加一个是否处于 pir 模式的分支判断，如果不处于 pir 模式下才去获取 y.name

**Review by MarioLulab — APPROVED** [review]
LGTM

File: test/legacy_test/test_smooth_l1_loss.py [hunk 8]
```diff
@@ -46,33 +47,40 @@ def setUp(self):
...
-                    'label': label_np,
-                },
-                fetch_list=[ret],
-            )
-            self.assertIsNotNone(static_ret)
+
+        expected = smooth_l1_loss_np(input_np, label_np, reduction='mean')
+
+        @test_with_pir_api
+        def test_dynamic_or_pir_mode():
```
Comment (0x45f) [inline test/legacy_test/test_smooth_l1_loss.py:60]: 这里的函数名是不是可以再修改一下？这个函数我看应该是测试静态图的，并没有测试动态图呀，是不是叫test_static好一点？

File: test/legacy_test/test_smooth_l1_loss.py [hunk 9]
```diff
@@ -81,41 +89,46 @@ def test_smooth_l1_loss_mean(self):
...
-                    'input': input_np,
-                    'label': label_np,
-                },
-                fetch_list=[ret],
-            )
-            self.assertIsNotNone(static_ret)
+        expected = smooth_l1_loss_np(input_np, label_np, reduction='sum')
+
+        @test_with_pir_api
+        def test_dynamic_or_pir_mode():
```
Comment (0x45f) [inline test/legacy_test/test_smooth_l1_loss.py:108]: 同上

File: test/legacy_test/test_smooth_l1_loss.py [hunk 10]
```diff
@@ -124,41 +137,46 @@ def test_smooth_l1_loss_sum(self):
...
-                    'input': input_np,
-                    'label': label_np,
-                },
-                fetch_list=[ret],
-            )
-            self.assertIsNotNone(static_ret)
+        expected = smooth_l1_loss_np(input_np, label_np, reduction='none')
+
+        @test_with_pir_api
+        def test_dynamic_or_pir_mode():
```
Comment (0x45f) [inline test/legacy_test/test_smooth_l1_loss.py:156]: 同上

File: test/legacy_test/test_smooth_l1_loss.py [hunk 11]
```diff
@@ -167,42 +185,47 @@ def test_smooth_l1_loss_none(self):
...
-                    'input': input_np,
-                    'label': label_np,
-                },
-                fetch_list=[ret],
-            )
-            self.assertIsNotNone(static_ret)
+        expected = smooth_l1_loss_np(input_np, label_np, delta=delta)
+
+        @test_with_pir_api
+        def test_dynamic_or_pir_mode():
```
Comment (0x45f) [inline test/legacy_test/test_smooth_l1_loss.py:205]: 同上

**Review by 0x45f — APPROVED** [review]
LGTM，关于函数名的修改可以另提一个PR进行修改~

**Pull request merged**